workloads/controllers/offline-virtual-machine.md

```diff
@@ -0,0 +1,265 @@
+
+### Starting and stopping
+
+After creating an OfflineVirtualMachine it can be switched on or off like this:
+
+```bash
+# Start the virtual machine:
+virtctl start myvm
+
+# Stop the virtual machine:
+virtctl stop myvm
+```
+
+`kubectl` can be used too:
+
+```bash
+# Start the virtual machine:
+kubectl patch offlinevirtualmachine myvm --type merge -p \
+    '{"spec":{"running":true}}'
+
+# Stop the virtual machine:
+kubectl patch offlinevirtualmachine myvm --type merge -p \
+    '{"spec":{"running":false}}'
+```
+
+### Controller status
+
+Once a VirtualMachine is created, it's state will be tracked via
+`status.created` and `status.ready`. If a VirtualMachine exists in the cluster,
+`status.created` will equal to `true`. If the VirtualMachine is also ready,
+`status.ready` will equal `true` too.
+
+If a VirtualMachine reaches a final state but the `spec.running` equals `true`,
+the OfflineVirtualMachine controller will set `status.ready` to `false` and
+re-create the VirtualMachine.
+
+### Restarting
+
+A VirtualMachine restart can be triggered by deleting the VirtualMachine. This
+will also propagate configuration changes from the template in the
+OfflineVirtualMachine:
+
+```bash
+# Restart the offline virtual machine (you delete the vm!):
+kubectl delete virtualmachine myvm
+```
+
+### Fencing considerations
+
+An OfflineVirtualMachine will never restart or re-create a VirtualMachine until
+the current instance of the VirtualMachine is deleted from the cluster.
+
+## When to use an OfflineVirtualMachine
+
+### When ABI stability is required between restarts
+
+An _OfflineVirtualMachine_ makes sure that VirtualMachine ABI configurations
+are consistent between restarts. A classical example are licenses which are
+bound to the firmware UUID of a virtual machine. The _OfflineVirtualMachine_
+makes sure that the UUID will always stay the same without the user having to
+take care of it.
+
+One of the main benefits is that a user can still make use of defaulting logic,
+although a stable ABI is needed.
+
+### When config updates should be picked up on the next restart
+
+If the VirtualMachine configuration should be modifyable inside the cluster and
+these changes should be picked up on the next VirtualMachine restart. This
+means that no hotplug is involved.
+
+### When you want to let the cluster manage your individual VirtualMachine
+
+Kubernetes as a declarative system can help you to manage the VirtualMachine.
+
+You tell it that you want this VirtualMachine with your application running,
+the OfflineVirtualMachine will try to make sure it stays running.
+
+> **Note**: The current believe is that if it is defined that the
+> VirtualMachine should be running, it should be running. This is different to
+> many classical virtualization platforms, where VMs stay down if they were
+> switched off. Restart policies may be added if needed. Please provide your
+> use-case if you need this!
+
+## Example
```

Comment:
should we just start linking to entries in kubvirt github's cluster/examples dir in our documentation?

Reply:
Hm, at the end they should be visible here. Did you have something in mind, so that we can then see the text here too, or really just a plain link?

Reply:
plain link or a docs build step that injects them.

This isn't something that needs to be addressed in this PR. I'm just worried we're creating something un-maintainable in our docs with all these examples.